File: Exercises/ExercisesFortran/11Jan_Chapter07/4-BarrierCrossing/Source/sample.f
      Subroutine Sample(Switch,Ttt)
      Implicit None

      Include 'system.inc'

      Integer  Maxx,I,Ttt,Switch
      Parameter(Maxx=50000)

      Double Precision Kt(Maxx),Ks(Maxx)
      Save Kt,Ks

      If (Switch.Eq.1) Then

C     Initialize Everything

         Do I=1,Maxx
            Kt(I) = 0.0d0
            Ks(I) = 0.0d0
         Enddo

      Elseif(Switch.Eq.2) Then

Ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
C     Compute the Transmission Coeffient                  C
C     see also Case Study 23 on page 440/441 (equation c) C
C                                                         C
C     The Maximum Time Equals Maxx                        C
Ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc

C     Begin Modification

C     End Modification

      Else

C     Write Results

         Do I=1,Maxx
            If(I.Le.Nstep) Then

               If(Ks(I).Gt.0.5d0) Write(27,*) Tstep*Dble(I-1),
     &            2.0d0*Kt(I)/Ks(I)

            Endif
         Enddo

      Endif

      Return
      End

File: Exercises/ExercisesC/11Jan_Chapter07/4-BarrierCrossing/Source/sample.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "system.h"

#define Maxx 50000

void Sample(int Switch, int Ttt)
{
  int i;
  static double Kt[Maxx],Ks[Maxx];
  FILE *FilePtr;

  switch(Switch)
  {
    case 1:
      // initialize everything
      for(i=0;i<Maxx;i++)
      {
        Kt[i]=0.0;
        Ks[i]=0.0;
      }
      break;
    case 2:
      // Compute the Transmission Coeffient
      // see also Case Study 23 on page 440/441 (equation c)
      // the maximum time equals Maxx

      // begin modification

      // end modification
      break;
    case 3:
      // write results
      FilePtr=fopen("results.dat","w");
      for(i=0;i<Maxx;i++)
        if(i<NumberOfSteps&&Ks[i]>0.5)
          fprintf(FilePtr,"%lf %lf\n",i*Tstep,2.0*Kt[i]/Ks[i]);
      fclose(FilePtr);
      break;
  }
}
